pkg/ddl/ingest/integration_test.go

```diff
@@ -379,3 +380,51 @@ func TestMultiSchemaAddIndexMerge(t *testing.T) {
 		tk.MustExec("admin check table t;")
 	}
 }
+
+func TestAddIndexIngestJobWriteConflict(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	defer ingesttestutil.InjectMockBackendMgr(t, store)()
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("create table t (a int primary key, b int);")
+	tk.MustExec("insert into t values (1, 1), (2, 2), (3, 3);")
+	tk.MustExec("set global tidb_enable_dist_task = off;")
+
+	injected := false
+	failpoint.EnableCall("github.com/pingcap/tidb/pkg/ddl/afterRunIngestReorgJob", func(job *model.Job, done bool) {
+		if done && !injected {
+			tk2 := testkit.NewTestKit(t, store)
+			tk2.MustExec("use test")
+			// Simulate write-conflict on ddl job table.
+			updateSQL := fmt.Sprintf("update mysql.tidb_ddl_job set processing = 0 where job_id = %d", job.ID)
+			tk2.MustExec(updateSQL)
+			updateSQL = fmt.Sprintf("update mysql.tidb_ddl_job set processing = 1 where job_id = %d", job.ID)
+			tk2.MustExec(updateSQL)
+			injected = true
+		}
+	})
+	rowCnt := atomic.Int32{}
+	failpoint.EnableCall("github.com/pingcap/tidb/pkg/ddl/ingest/onMockWriterWriteRow", func() {
+		rowCnt.Add(1)
+	})
+	tk.MustExec("alter table t add index idx(b);")
+	require.True(t, injected)
+	// Write conflict error should not retry the whole job.
+	require.Equal(t, 3, int(rowCnt.Load())) // it should not be 6
+	tk.MustExec("admin check table t;")
+}
+
+func TestAddIndexIngestPartitionCheckpoint(t *testing.T) {
```

Comment:
do we have something to check in this test?

Reply:
https://github.com/pingcap/tidb/pull/54747/commits/eaa11ed4af79fa91bc6fce6548b879becb4f2549